napalm_junos/junos.py

```diff
@@ -1372,13 +1372,16 @@ def get_config(self, retrieve='all'):
 
         if retrieve in ('candidate', 'all'):
             config = self.device.rpc.get_config(filter_xml=None, options=options)
-            rv['candidate'] = py23_compat.text_type(config.text.encode('ascii', 'replace'))
+            rv['candidate'] = py23_compat.text_type(config.text)
+#            rv['candidate'] = py23_compat.text_type(config.text.encode('ascii', 'replace'))
 
         if retrieve in ('running', 'all'):
             options['database'] = 'committed'
             config = self.device.rpc.get_config(filter_xml=None, options=options)
-            rv['running'] = py23_compat.text_type(config.text.encode('ascii', 'replace'))
+            rv['running'] = py23_compat.text_type(config.text)
```

Comment:
@mirceaulinic Do you see any reason we need the .encode('ascii', 'replace') here?

config.text is already a 'str' in Python2.

The encode('ascii', 'replace') causes an issue in Python3 in that it gets me back a byte-string and not a Python3 unicode string (i.e. I would have to go and decode the string).